.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,92 @@
+# Copyright 2021 Agnostiq Inc.
+#
+# This file is part of Covalent.
+#
+# Licensed under the GNU Affero General Public License 3.0 (the "License").
+# A copy of the License may be obtained with this software package or at
+#
+#      https://www.gnu.org/licenses/agpl-3.0.en.html
+#
+# Use of this file is prohibited except in compliance with the License. Any
+# modifications or derivative works of this file must retain this copyright
+# notice, and modified files must contain a notice indicating that they have
+# been altered from the originals.
+#
+# Covalent is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the License for more details.
+#
+# Relief from the License may be granted by purchasing a commercial license.
+
+name: benchmark
+
+on:
+  workflow_call:
+  workflow_dispatch:
+    inputs:
+      commit_sha:
+        description: "Commit SHA used for testing"
+        type: string
+        required: true
+
+jobs:
+  benchmarks:
+    runs-on: ubuntu-latest
+    env:
+      COVALENT_SERVER_IFACE_ANY: true
+    steps:
+      - name: Check out head
+        if: github.event_name == 'workflow_call'
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+
+      - name: Check out SHA
+        if: github.event_name == 'workflow_dispatch'
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+          fetch-depth: 0
+          ref: ${{ inputs.commit_sha }}
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.8"
+
+      - name: Install Covalent
+        run: pip install -e .
```

Comment:
what's the right way to install Covalent in CI environments? some workflows have `-e` and some don't

Reply:
yes please

Reply:
one more thought... @wjcunningham7 should this be built first and then installed from the dist?

Reply:
yes, I agree the optimal way to install in CI is with `python setup.py sdist && pip install dist/*.tar.gz`